src/Data/Long/FFI.erl

```diff
@@ -0,0 +1,110 @@
+-module(data_long_FFI@foreign).
+
+-export([ add/1
+        , 'and'/1
+        , compare
+        , divide/1
+        , equals/1
+        , divide/1
+        , equals/1
+        , getHighBits/1
+        , getHighBitsUnsigned/1
+        , getLowBits/1
+        , getLowBitsUnsigned/1
+        , getNumBitsAbs/1
+        , greaterThan/1
+        , greaterThanOrEqual/1
+        , isEven/1
+        , idOdd/1
+        , isPositive/1
+        , isZero/1
+        , lessThan/1
+        , lessThanOrEqual/1
+        , modulo/1
+        , multiply/1
+        , negate/1
+        , 'not'/1
+        , notEquals/1
+        , 'or'/1
+        , shiftLeft/1
+        , shiftRight/1
+        , shiftRightUnsigned/1
+        , rotateLeft/1
+        , rotateRight/1
+        , subtract /1
+        , toBytes/1
+        , toInt/1
+        , toNumber/1
+        , toSigned/1
+        , toString/1
+        , toUnsigned/1
+        , 'xor'/1
+        ]
+  )
+
+add(A) -> fun(B) -> A + B end.
```

Comment:
in purerl FFI you an also do this:

```erlang
add(A,B) -> A + B end.
```

```
foreign import add :: Int -> Int -> Int
```

will use the arity-2 version or the arity-1

Reply:
ah nice didn't know that, not entirely sure if the plan is to stick a PR upstream see if the original contributor will take it or to keep our fork if the later then I'll change them all.

Reply:
Your comment doesn't seem to bear any relevance to my point, I'm talking entirely about what you do in the `.erl`

Generally no ffi forks have been upstreamed, so probably wouldn't bother doing that, it';s not clear if the fork model will be the way forward in future - some other forks are doing a "sideloading" model which could be lower maintenance, but I'm not sure entirely works for me